main/src/main/scala/sbt/plugins/SbtPlugin.scala

```diff
@@ -10,11 +10,14 @@ package plugins
 
 import Keys._
 import Def.Setting
+import sbt.SlashSyntax0._
+import sbt.librarymanagement.Configurations.Compile
 
 object SbtPlugin extends AutoPlugin {
   override def requires = ScriptedPlugin
 
   override lazy val projectSettings: Seq[Setting[_]] = Seq(
-    sbtPlugin := true
+    sbtPlugin := true,
+    Compile / scalacOptions += "-Wconf:cat=unused-nowarn:s"
```

Comment:
Isn't this too global? Wouldn't this silence also other "unused" warnings?

Reply:
This option does only silence the unused `@nowarn` warnings in the `SbtPlugin` projects.

```
@nowarn annotation does not suppress any warnings
```

It is not a perfect solution because it will silence your own unused `@nowarn` annotations. But it is very specific to sbt plugins and it is harmless.

A better solution would be to have scala/scala#9491 but it is only available in 2.13.